pkg/kubelet/eviction/eviction_manager.go

```diff
@@ -197,20 +197,21 @@ func (m *managerImpl) synchronize(diskInfoProvider DiskInfoProvider, podFunc Act
 
 	glog.V(3).Infof("eviction manager: synchronize housekeeping")
 
+	hasDedicatedImageFs, errImage := diskInfoProvider.HasDedicatedImageFs()
 	// build the ranking functions (if not yet known)
 	// TODO: have a function in cadvisor that lets us know if global housekeeping has completed
 	if len(m.resourceToRankFunc) == 0 || len(m.resourceToNodeReclaimFuncs) == 0 {
 		// this may error if cadvisor has yet to complete housekeeping, so we will just try again in next pass.
-		hasDedicatedImageFs, err := diskInfoProvider.HasDedicatedImageFs()
-		if err != nil {
+		if errImage != nil {
```

Comment:
From the existing code, I think the logic is if resourceToRankFunc and resourceToNodeReclaimFuncs are already defined, it does not matter whether errImage returns nil or not. 
I modified the code here a little more. Instead of keeping calling HasDedicatedImageFs in every sync call, I put it in a map. PTAL